### ISSUE TITLE
add formatting reset after gecos in cliconn snotes

### ISSUE THREAD

**Issue by jesopo** [solanum-ircd/solanum#87]

**Comment by dwfreed** [comment]
Perhaps we should just alter a user's provided gecos to end with a format reset if it includes any control codes?  Another option would be to just strip formatting from gecos.

**Comment by aaronmdjones** [comment]
I would very much prefer an option to strip formatting alltogether.

**Comment by jesopo** [comment]
strip it from gecos in general or strip it from gecos in cliconns?

also, can't force a reset char on people if their gecos is already the max length

**Comment by dwfreed** [comment]
> strip it from gecos in general or strip it from gecos in cliconns?
> 
> also, can't force a reset char on people if their gecos is already the max length

I'm saying eat their last char if their gecos is max length and give them a reset char no matter what (if they used control codes in their gecos).  One character won't make or break their gecos.

**Comment by aaronmdjones** [comment]
> strip it from gecos in general or strip it from gecos in cliconns?

A `general {}` option to strip it on client registration.

**Comment by ilbelkyr** [comment]
> One character won't make or break their gecos.

one *byte*, fwiw, might end up cutting off part of a multibyte character – granted, this can already happen if clients specify a realname longer than the maximum length.

**Comment by glguy** [comment]
I don't think we should alter people's GECOS beyond truncation. I'd prefer rejecting invalid values to mangling. If we find clients that break when the GECOS field has unterminated formatting we should file bugs/PRs for those clients.

I would want the snote to exactly match what the client's GECOS field actually was. I might need to rely on that for other kinds of pattern matching. Thus I'm less interested in always or conditionally adding a ^O terminator to that cliconn note.

**Comment by xnrand** [comment]
A downside of stripping is that a connection could end up with an empty gecos, which is something that can't happen otherwise.

Another option would be to reject connections with invalid gecoses entirely, similarly to how connections with invalid characters in the username are rejected, rather than those characters being stripped.
